src/backend/tools/google_drive.py

```diff
@@ -0,0 +1,162 @@
+import datetime
+import json
+import os
+import urllib.parse
+from typing import Any, Dict, List
+
+import requests
+from fastapi import Request
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+
+from backend.crud import tool_auth as tool_auth_crud
+from backend.database_models.database import DBSessionDep
+from backend.database_models.tool_auth import ToolAuth
+from backend.services.logger import get_logger
+from backend.tools.base import BaseAuth, BaseTool
+
+GOOGLE_DRIVE_TOOL_ID = "google_drive"
+logger = get_logger()
+
+
+class GoogleDrive(BaseTool):
+    """
+    Experimental (In development): Tool that searches Google Drive
+    """
+
+    @classmethod
+    def is_available(cls) -> bool:
```

Comment:
Any chance we can check if the user is authenticated here?

Reply:
We could but I think it is better to have the error in is_auth_required?